### ISSUE TITLE
shrink(1) changes multiclass predictions instead of preserving the first tree (CatBoostClassifier 1.2.8 and 1.2.10)

### ISSUE THREAD

**Issue by eseglo** [catboost/catboost#3136]
I believe there is a bug in CatBoostClassifier.shrink() for multiclass classification.

Calling model.shrink(1) changes the model predictions to values that do not correspond to the predictions of the first tree. Instead, the predictions become almost constant and close to the class priors.

This issue reproduces with both CatBoost 1.2.8 and 1.2.10.

The issue was discovered while debugging AutoGluon, but it can be reproduced with plain CatBoost without any external libraries.

Minimal reproducer
```
import pandas as pd
from catboost import CatBoostClassifier, Pool

# Simple categorical dataset
X_train = pd.DataFrame({
    "X0": pd.Categorical(
        ["A"] * 679 +
        ["B"] * 97 +
        ["C"] * 14 +
        ["D"] * 178
    )
})

y_train = (
    ["C"] * 268 + ["B"] * 223 + ["A"] * 188 +
    ["A"] * 58 + ["B"] * 27 + ["C"] * 12 +
    ["A"] * 13 + ["B"] * 1 +
    ["A"] * 151 + ["B"] * 22 + ["C"] * 5
)

train_pool = Pool(
    X_train,
    y_train,
    cat_features=["X0"]
)

model = CatBoostClassifier(
    iterations=10000,
    learning_rate=0.05,
    random_seed=0,
    verbose=False,
)

model.fit(train_pool)

X_test = pd.DataFrame({
    "X0": pd.Categorical(["A", "B", "C", "D"])
})

print("Predictions before shrink")
print(model.predict_proba(X_test))
print("Tree count:", model.tree_count_)

model.shrink(1)

print("\nPredictions after shrink(1)")
print(model.predict_proba(X_test))
print("Tree count:", model.tree_count_)
```
Output (CatBoost 1.2.10)
Predictions before shrink
[[5.84157621e-04 2.48780121e-01 7.50635721e-01]
 [9.99047184e-01 7.02300586e-04 2.50515780e-04]
 [9.98720295e-01 1.09334729e-03 1.86357529e-04]
 [8.75461557e-01 7.57949184e-04 1.23780494e-01]]

Tree count: 10000

Predictions after shrink(1)
[[0.32078199 0.32617316 0.35304484]
 [0.35878235 0.32170180 0.31951584]
 [0.35878235 0.32170180 0.31951584]
 [0.35878235 0.32170180 0.31951584]]

Tree count: 1
Expected behavior

After

model.shrink(1)

the model should behave identically to a model consisting of only the first tree.

Predictions should therefore correspond to the first boosting iteration, not collapse to nearly constant probabilities.

Actual behavior

After shrink(1):

tree_count_ correctly becomes 1
predictions change dramatically
most inputs receive nearly identical probability vectors
the outputs resemble class-prior probabilities rather than the contribution of the first tree
Additional observations

While investigating this, I also observed that:

Training normally (without validation) works correctly.
Training with an eval_set and use_best_model=True produces exactly the same incorrect predictions as calling shrink(best_iteration + 1) afterwards.
If use_best_model=False, predictions are correct.
Therefore the issue appears to originate from the internal shrinking logic rather than from early stopping itself.

This suggests that shrink() is not preserving the ensemble state correctly for multiclass models.

Environment
CatBoost 1.2.8
CatBoost 1.2.10
Python 3.10
Linux x86_64

**Comment by adityaanikam** [comment]
I looked into this. `shrink(1)` goes through `core.py`'s `shrink()` → `_base_shrink()` in `_catboost.pyx` → `TFullModel::Truncate()` in `catboost/libs/model/model.h`:

```cpp
void Truncate(size_t begin, size_t end) {
     ModelTrees.GetMutable()->TruncateTrees(begin, end);
     if (CtrProvider) { CtrProvider->DropUnusedTables(applyData->GetUsedModelCtrBases()); }
     if (begin > 0) {
         SetScaleAndBias({GetScaleAndBias().Scale, {}});
     }
     UpdateDynamicData();
 }
 ```

Bias is only reset when `begin > 0`. `shrink(1)` calls `Truncate(0, 1)`, so that branch never runs and the model keeps its original bias untouched. I traced where that bias comes from in `train_model.cpp`: `SetScaleAndBias({1, ctx.LearnProgress->StartingApprox...})` it's the pre training baseline (for multiclass, effectively the class prior logits). A 1 tree model still carrying the full ensemble's bias unchanged would plausibly explain predictions collapsing toward class priors, since that's what the retained bias represents.

I haven't been able to compile the C++ side here to actually confirm this is the differential cause versus a natively trained 1 tree model (which presumably starts from the same bias and would need to behave the same way for this theory to hold) so this is a hypothesis from reading the code, not a verified fix. Wanted to raise it here before attempting a PR, per the contributing guide. Does the `begin > 0` condition look intentional, or should bias reset happen on any truncation where trees are dropped?
